### ISSUE TITLE
PlotData broken?

### ISSUE THREAD

**Issue by asoplata** [DynaSim/DynaSim#143]
I've been trying to run the default `demos.m` script in the hopes we can use its examples to build our tests tomorrow (which was @erik-roberts idea), but the third example seems to fail due to what I think is an almost two-month-old bug in `PlotData()`, and a big one. I've seen this bug on `master`, `dev`, and the original commit where the `PlotData()` code change was pushed, https://github.com/DynaSim/DynaSim/commit/4fd131936786e12adab113aa5fb5176bee05aa5c .

On [Line 130](https://github.com/DynaSim/DynaSim/blob/dev/demos/demos.m#L130) of `demos.m`, it makes a trivial call to `PlotData()` for a simulation with 8 "varies". However, I get an error from `PlotData()` [Line 186](https://github.com/DynaSim/DynaSim/blob/dev/functions/PlotData.m#L186) that its attempt to reshape `data` using JUST the dimensions that are ACTUALLY different between variations fails. This is only a problem if the number of variations is greater than 2, since that's the if block it's in in `PlotData()`.

```{matlab}
Error using reshape
To RESHAPE the number of elements must not change.

Error in PlotData (line 186)
        data = reshape(data, prod(effective_vary_lengths(3:end)), prod(effective_vary_lengths([1 2])));

Error in demos (line 130)
PlotData(data);
```

I tried searching the Issues for prior mention but didn't see anything, and I'm wondering if there's just something wrong with my setup (I extensively cleaned my path before investigating this). Has `PlotData()` been officially deprecated in favor of `PlotData2()`, or is this a problem on my part, or is there a bug that `PlotData()` doesn't work on simulations that have more than 2 vary dimensions? Can someone else test this and confirm?

Somewhat relatedly, has anyone actually been using all this new code we're pushing to do real simulation work? I definitely haven't. If no one has, then testing is THAT much more important, it's a dang good thing we're planning on writing tests tomorrow!!!

**Comment by asoplata** [comment]
Also, if this is difficult to fix, it may be a good reason for making `PlotData2()` the "default" for DynaSim.

**Comment by benpolletta** [comment]
I doubt this is difficult to fix. I'll check it out later this morning. I've used PlotData to plot things with more than two vary *dimensions* since these changes were made, so it was working at some point on some use cases.

**Comment by davestanley** [comment]
I just took a look at the simulation output here.

It seems the issue is that, although they are only 8 varies, the simulation calls are exploring a very high dimensional (5D) parameter space. I guess we don't really make use of this use case often, so it's not surprising to me that a bug crept in to PlotData.

**Comment by davestanley** [comment]
I tried running this in PlotData2 and it broke also.

Is this a use case that we actually want to support? It seems rather specialized (e.g. instead of doing sweeps, we only look at a few specific points in high dimensional parameter space)

PlotData2 breaks because, although this space is sparsely filled, it's still a 5D hypercube, which PlotData2 tries to represent explicitly. I wrote a work-around to plot this, but it wasn't pretty. 

The DynaSim data structure produced by the simulation is as follows (note: I ran all of this on my branch, bc1f8044, not the main dev branch)

Running this code produces the following
```
%%
xp1 = All2xPlt(data);
xp1 = xp1.mergeDims([1:5]);   % This is slow....
xp1.getaxisinfo

xp1 = xp1.squeezeRegexp('Dim');
xp1.getaxisinfo

emptys = cellfun(@isempty,xp1.data(:,:,1,1));

xp1 = xp1(~emptys,:,:);

data1 = xPlt2DynaSim(xp1);

PlotData2(data1,'variable','v');
%PlotData2(data1,'variable','u','lock_axes',0);
```

![image](https://cloud.githubusercontent.com/assets/5439305/24835411/d881b034-1ccf-11e7-9612-052febcf1142.png)

Or you can get a 2D space representation as follows

```
xp2 = All2xPlt(data);
xp2 = xp2.mergeDims([1:3]); xp2 = xp2.mergeDims([4:5]);
xp2.getaxisinfo

xp2 = xp2.squeezeRegexp('Dim');
xp2.getaxisinfo


emptymat = cellfun(@isempty,xp2.data(:,:,1,1));
figure; imagesc(emptymat)

empty_rows = all(emptymat,2);
empty_cols = all(emptymat,1);

xp2 = xp2(find(~empty_rows),find(~empty_cols),:,:);


data2 = xPlt2DynaSim(xp2);

PlotData2(data2)
```

![image](https://cloud.githubusercontent.com/assets/5439305/24835456/fd878d26-1cd0-11e7-912f-71d859bc2345.png)

**Comment by erik-roberts** [comment]
this use case is important from a simulating standpoint for filling in missing sims after the fact and doing iterative sampling. having all the plots in one panel like that in plot data seems unnecessary, and gimbal-vis can handle that easily, albeit you have to scroll over them.

**Comment by benpolletta** [comment]
The problem in PlotData isn't that it's high-dimensional. The problem is
the function that checks for covaried parameters. I wrote it assuming that
if two parameters are covaried, the values of each parameter will change
over all simulations. Here, though, there are several parameters whose
values are the same over multiple simulations. That's what breaks PlotData,
and I think it's something worth fixing since it's a pretty general use
case. No bright ideas yet but I'm thinking about it.

**Comment by davestanley** [comment]
The specific use case is "visualization of a sparse high dimensional parameter space". IMO it could fall into the domain of either gimble-vis or PlotData. 

I can fairly easily incorporate this option into PlotData2 if it would be useful.

**Comment by benpolletta** [comment]
Fixed as of 627108c1acfb6b972f1c8978602ec79e993817b4.
